src/js/common/array.js

```diff
@@ -221,45 +224,155 @@ function stringDESCIgnoreCase(_a, _b) {
 }
 
 /**
- * Compare schedule models for sort.
+ * Compare schedule models included a travel time for sort.
  *
  * 1. all day schedule first.
- * 2. early start.
+ * 2. early start included a travel time.
  * 3. longest duration.
  * 4. early created.
  * @param {Schedule|ScheduleViewModel} a The object schedule instance.
  * @param {Schedule|ScheduleViewModel} b The object schedule instance.
  * @returns {number} Result of comparison.
  */
 function scheduleASC(a, b) {
-    var durationA, durationB;
-    var allDayCompare, startsCompare;
+    var allDayCompare, startsCompare, durationCompare;
     var modelA = a.valueOf();
     var modelB = b.valueOf();
 
-    allDayCompare = booleanASC(modelA.isAllDay || a.hasMultiDates, modelB.isAllDay || b.hasMultiDates);
+    allDayCompare = _getAllDayCompare(a, b);
 
     if (allDayCompare) {
         return allDayCompare;
     }
 
-    startsCompare = datetime.compare(a.getStarts(), b.getStarts());
+    startsCompare = _getStartCompare(a, b);
 
     if (startsCompare) {
         return startsCompare;
     }
 
-    durationA = a.duration();
-    durationB = b.duration();
+    durationCompare = _getDurationCompare(a, b);
+
+    if (durationCompare) {
+        return durationCompare;
+    }
+
+    return util.stamp(modelA) - util.stamp(modelB);
+}
+
+/**
+ * Compare events in multiple dates or all-day events
+ *
+ * @param {Schedule|ScheduleViewModel} a The object schedule instance.
+ * @param {Schedule|ScheduleViewModel} b The object schedule instance.
+ * @returns {number} Result of comparison.
+ */
+function _getAllDayCompare(a, b) {
+    return booleanASC(a.valueOf().isAllDay || a.hasMultiDates, b.valueOf().isAllDay || b.hasMultiDates);
+}
+
+/**
+ * Calculate start time, including the going duration of travel time
+ *
+ * @param {Schedule|ScheduleViewModel} a The object schedule instance.
+ * @param {Schedule|ScheduleViewModel} b The object schedule instance.
+ * @returns {number} Result of comparison.
+ */
+function _getStartCompare(a, b) {
+    var aTravelTime = getMaxTravelTime(a);
+    var bTravelTime = getMaxTravelTime(b);
+    var aGoingDuration = datetime.millisecondsFrom('minutes', aTravelTime.maxGoingDuration);
+    var bGoingDuration = datetime.millisecondsFrom('minutes', bTravelTime.maxGoingDuration);
+    var renderedStartA = new TZDate(a.getStarts().getTime() - aGoingDuration);
```

Comment:
Why do you use TZDate? I think just plus and minus are enough.

Reply:
When passing as a parameter to `datetime.compare`, the function requires a `Date` object.

```js
compare: function(d1, d2) {
   var _d1 = d1.getTime(),
       _d2 = d2.getTime();
...
```

Would it be better to make it a Date object when calling the compare function?

```js
return datetime.compare(new TZDate(renderedStartA), new TZDate(renderedStartB));
```